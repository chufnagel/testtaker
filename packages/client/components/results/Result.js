import React, { memo } from "react";
import PropTypes from "prop-types";

import Passage from "../Passage";
import ProgressBar from "../ProgressBar";
import QuestionAnswerCard from "./QuestionAnswerCard";

const Result = memo(
  ({ passage, questions, totalCorrect, totalQuestions, elapsedTime, totalTime }) => {
    const correct = (totalCorrect / totalQuestions) * 100;
    const elapsed = (elapsedTime / totalTime) * 100;
    let renderPassage;
    if (passage) {
      renderPassage = <Passage passage={passage} />;
    } else {
      renderPassage = <div />;
    }
    return (
      <div>
        <h1>Results</h1>
        <ProgressBar percentage={correct} />
        <p>
          {totalCorrect} out of {totalQuestions} correct
        </p>
        <ProgressBar percentage={elapsed} />
        <p>
          {elapsedTime} out of {totalTime} used
        </p>
        {renderPassage}
        {questions.map(question => (
          <QuestionAnswerCard key={question.id} {...question} />
        ))}
      </div>
    );
  },
);

Result.propTypes = {
  // questions: PropTypes.arrayOf(PropTypes.object).isRequired,
  questions: PropTypes.arrayOf(PropTypes.object),
  // answers: PropTypes.arrayOf(PropTypes.object).isRequired,
  // correctAnswers: PropTypes.arrayOf(PropTypes.object).isRequired,
  passage: PropTypes.arrayOf(PropTypes.object),
  totalCorrect: PropTypes.number,
  totalQuestions: PropTypes.number,
  elapsedTime: PropTypes.number,
  totalTime: PropTypes.number,
};

Result.defaultProps = {
  questions: [
    {
      id: 1,
      text: "Navita (line 2) refers to",
      correct: "A",
      userAnswer: "B",
      answers: [
        {
          choice: "A",
          content: "Evander",
        },
        {
          choice: "B",
          content: "Latinus",
        },
        {
          choice: "C",
          content: "Priam",
        },
        {
          choice: "D",
          content: "Juturna",
        },
      ],
    },
  ],
  passage: [
    {
      line: "1",
      text: "Ergo iter inceptum peragunt fluvioque propinquant.",
    },
  ],
  totalCorrect: 15,
  totalQuestions: 20,
  elapsedTime: 15,
  totalTime: 15,
};

export default Result;
