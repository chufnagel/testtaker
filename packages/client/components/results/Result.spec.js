/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import Result from "./Result";

// Result.propTypes = {
//   questions: PropTypes.arrayOf(PropTypes.object).isRequired,
//   answers: PropTypes.arrayOf(PropTypes.object).isRequired,
//   correctAnswers: PropTypes.arrayOf(PropTypes.object).isRequired,
//   totalCorrect: PropTypes.number,
//   totalQuestions: PropTypes.number,
//   elapsedTime: PropTypes.number,
//   totalTime: PropTypes.number,
// };

describe("Result", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Result />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("Result matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
