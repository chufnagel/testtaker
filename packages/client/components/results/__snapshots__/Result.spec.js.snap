// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Result matches snapshot 1`] = `
<div>
  <h1>
    Results
  </h1>
  <ProgressBar
    percentage={75}
  />
  <p>
    15
     out of 
    20
     correct
  </p>
  <ProgressBar
    percentage={100}
  />
  <p>
    15
     out of 
    15
     used
  </p>
  <Passage
    passage={
      Array [
        Object {
          "line": "1",
          "text": "Ergo iter inceptum peragunt fluvioque propinquant.",
        },
      ]
    }
  />
  <QuestionAnswerCard
    answers={
      Array [
        Object {
          "choice": "A",
          "content": "Evander",
        },
        Object {
          "choice": "B",
          "content": "Latinus",
        },
        Object {
          "choice": "C",
          "content": "Priam",
        },
        Object {
          "choice": "D",
          "content": "Juturna",
        },
      ]
    }
    correct="A"
    id={1}
    key="1"
    text="Navita (line 2) refers to"
    userAnswer="B"
  />
</div>
`;
